update documentation for ws-socket and vault security

sichen1234 opened the pull request:
@brioux @Zzocker Can you please review these changes to the documentation based on our call on Monday?

I'm still not too clear where the ws-identity and ws-identity-client are vs each other. Does ws-wallet talk to ws-identity, which then talks through ws-identity-client to Fabric?

A client application that uses web socket identity then somehow incorporate ws-wallet in its UI?

brioux commented:
@sichen1234 


I have cloned your the opentaps main branch and added some changes on [my own branch](https://github.com/brioux/blockchain-carbon-accounting/tree/opentaps-main).
1. I have corrected the secure-identities README for ws-identity-client. It is only used as a backend connector for the Fabric app (not ws-wallet) to communicate with ws-identity. it is just the rpc used by the fabric app to reach the server and send signature requests the the Users ws-wallet.
2. A client application that uses web socket identity then somehow incorporate ws-wallet in its UI? Yes! it contacts the server using the provided ws:// url
3.  I have added secure-identities/docs/ with a detailed ECB diagram for setting up a ws-identity server session between the User, its ws-wallet, and the Fabric app: Note this includes two controllers: the web-socket-client setup by the Users ws-wallet, and the corresponding web-socket-client setup on the ws-identity server.

brioux reviewed (changes requested):
@sichen1234 
I have cloned the branch in this PR and modified it on [my own branch](https://github.com/brioux/blockchain-carbon-accounting/tree/opentaps-main) (I cannot edit your PR)
1. Regrding your question 

> Does ws-wallet talk to ws-identity, which then talks through ws-identity-client to Fabric?

Not exactly, I have corrected the secure-identities README for ws-identity-client. It is only used as a backend connector for the Fabric app (not ws-wallet) to talk to ws-identity. I.e., it setups the rpc to request session tickets and send signature requests to the ws-wallet of the connected User. _Note: This could be replaced by any other promise based HTTP client (e.g. [axios](https://www.npmjs.com/package/axios))_

2.  I have added [secure-identities/docs/ with a detailed ECB diagram](https://github.com/brioux/blockchain-carbon-accounting/blob/opentaps-main/secure-identities/docs/ws-session.md)for setting up ws-identity server sessions between the User, its ws-wallet, and the Fabric app. This includes a web-socket-client controller pair: one setup for the Users ws-wallet, and another for the ws-identity server.

Regarding your question -

> A client application that uses web socket identity then somehow incorporate ws-wallet in its UI?

 Yes, and it contacts the ws-identity server using the generated ws:// url. Note, if the client app is not built in typescript an alternative/equivalent to ws-wallet is needed

sichen1234 commented:
Could you please create a PR from your branch then?  I will merge it in
along with my changes.
-----
Si Chen
Open Source Strategies, Inc.

New Video: Creating Climate Solutions with Blockchain and Open Source
<https://youtu.be/gtO3COq8crQ>


On Sat, Oct 30, 2021 at 3:11 PM Bertrand Rioux ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> @sichen1234 <https://github.com/sichen1234>
> I have cloned the branch in this PR and modified it on my own branch
> <https://github.com/brioux/blockchain-carbon-accounting/tree/opentaps-main>
> (I cannot edit your PR)
>
>    1. I have corrected the secure-identities README for
>    ws-identity-client. It is only used as a backend connector for the Fabric
>    app (not ws-wallet). I.e., it provides the rpc used by fabric app to reach
>    ws-identity server and send signature requests to the Users ws-wallet. It
>    is not essential as this can be achieve using any promise based HTTP client
>    (e.g. axios <https://www.npmjs.com/package/axios>)
>    2. I have added secure-identities/docs/ with a detailed ECB diagram
>    for ws-identity server sessions between the User, its ws-wallet, and the
>    Fabric app. This includes the web-socket-client controller pair: one setup
>    for the Users ws-wallet, and another for the ws-identity server.
>    Regarding you question -
>
> A client application that uses web socket identity then somehow
> incorporate ws-wallet in its UI?
>
> Yes, and contacts the ws-identity server using the generated ws:// url.
> Note, if the client app is not built in typescript an
> alternative/equivalent to ws-wallet must be built.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyperledger-labs/blockchain-carbon-accounting/pull/336#pullrequestreview-793630663>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANAS4LA7HST3ZL6ACCQGALUJRUPLANCNFSM5G6DANBA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

brioux commented:
@sichen1234 see [PR 337](https://github.com/hyperledger-labs/blockchain-carbon-accounting/pull/337)

univan0928 commented:
Hi.
Could you please let me know more details via skype?
Thank you.
James
live:.cid.9307a504597f0f67
llive:.cid.9307a504597f0f67ivlive:.cid.9307a504597f0f67
e:.cid.9307a504597f0f67

On Tue, Nov 2, 2021 at 9:20 PM Bertrand Rioux ***@***.***>
wrote:

> @sichen1234 <https://github.com/sichen1234> see PR 337
> <https://github.com/hyperledger-labs/blockchain-carbon-accounting/pull/337>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyperledger-labs/blockchain-carbon-accounting/pull/336#issuecomment-957556228>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATYPWTBR4QWRWHBQFMBA5T3UJ7QQVANCNFSM5G6DANBA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

univan0928 commented:
My skype name is univan0928.

On Tue, Nov 2, 2021 at 9:39 PM Univan ***@***.***> wrote:

> Hi.
> Could you please let me know more details via skype?
> Thank you.
> James
> live:.cid.9307a504597f0f67
> llive:.cid.9307a504597f0f67ivlive:.cid.9307a504597f0f67
> e:.cid.9307a504597f0f67
>
> On Tue, Nov 2, 2021 at 9:20 PM Bertrand Rioux ***@***.***>
> wrote:
>
>> @sichen1234 <https://github.com/sichen1234> see PR 337
>> <https://github.com/hyperledger-labs/blockchain-carbon-accounting/pull/337>
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/hyperledger-labs/blockchain-carbon-accounting/pull/336#issuecomment-957556228>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ATYPWTBR4QWRWHBQFMBA5T3UJ7QQVANCNFSM5G6DANBA>
>> .
>> Triage notifications on the go with GitHub Mobile for iOS
>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android
>> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>>
>>
>
